src/cross_module_fn.c

```diff
@@ -398,9 +415,10 @@ TSDLLEXPORT CrossModuleFunctions ts_cm_functions_default = {
 	/* compression */
 	.compressed_data_send = error_no_default_fn_pg_community,
 	.compressed_data_recv = error_no_default_fn_pg_community,
-	.compressed_data_in = error_no_default_fn_pg_community,
+	.compressed_data_in = process_compressed_data_in,
```

Comment:
I can't find a comment anywhere about what "compressed_data_in" is supposed to do.

This change is a little unclear, so you want to still have some wrapper for this function even in an unlicensed version, because the replication worker doesn't load TSL by default?

Reply:
This will work only in licensed version, unlicensed still will produce an error.

The problem was, that even in licensed version TSL module did not loaded properly from replication worker, so it produced misleading error which says that your license is "timescale" and you should upgrade to "timescale". I did not found a better place to check if module is loaded.

`compressed_data_in` function called when replication worker started to insert data to table created on dest data node. I think there is a type exists for compressed data `_timescaledb_internal.compressed_data`

Reply:
Ah, OK, I think I understand better now.
I'd write something like this:
```
/*
 * TSL library is not loaded by the replication worker for some reason,
 * so a call to `compressed_data_in` function would produce a misleading
 * error saying that your license is "apache" and you should upgrade to
 * "timescale" license, even if you have already upgraded.
 * As a workaround, we try to load the TSL module it in this function. It will still
 * error out in the "apache" version
 */
 ```
 
 And I'd also replace the function pointer check with `ts_is_license_apache`, it was kinda confusing for me.